### PR TITLE
Mobile text message rendering

### PR DESCRIPTION
**Pull request by chromakode**
The various states are all working. Not going to worry about message wrapping yet -- will look into that next w/ markup formatting.

:eyeglasses: @keybase/react-hackers

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits ea373fcd7ac9...222eaed75d9e introduce visual changes on linux. <details><summary>:mag_right: 2 new, 7 changed</summary>

 * Added: **Text_Message-edited**  
   Text_Message-edited rendered
 * Added: **Stacked_Text_Message-Stacked_two_messages_one_edited**  
   Stacked_Text_Message-Stacked_two_messages_one_edited rendered
 * Changed: **Text_Message-failed_You** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Text_Message-failed_You-ea373fcd7ac9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Text_Message-failed_You-222eaed75d9e.png)  
   Text_Message-failed_You comparison
 * Changed: **Text_Message-failed_Not_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Text_Message-failed_Not_Following-ea373fcd7ac9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Text_Message-failed_Not_Following-222eaed75d9e.png)  
   Text_Message-failed_Not_Following comparison
 * Changed: **Text_Message-failed_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Text_Message-failed_Following-ea373fcd7ac9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Text_Message-failed_Following-222eaed75d9e.png)  
   Text_Message-failed_Following comparison
 * Changed: **Text_Message-failed_Broken** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Text_Message-failed_Broken-ea373fcd7ac9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Text_Message-failed_Broken-222eaed75d9e.png)  
   Text_Message-failed_Broken comparison
 * Changed: **Stacked_Text_Message-Stacked_one_sent_one_failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-ea373fcd7ac9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-222eaed75d9e.png)  
   Stacked_Text_Message-Stacked_one_sent_one_failed comparison
 * Changed: **Attachment_Message-Basic_Preview_Image_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Attachment_Message-Basic_Preview_Image_Failed-ea373fcd7ac9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Attachment_Message-Basic_Preview_Image_Failed-222eaed75d9e.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Attachment_Message-Basic_Preview_Image_Failed.png)
 * Changed: **Attachment_Message-Basic_Generic_File_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Attachment_Message-Basic_Generic_File_Failed-ea373fcd7ac9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Attachment_Message-Basic_Generic_File_Failed-222eaed75d9e.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487042584521-ea373fcd7ac9-222eaed75d9e-linux/Attachment_Message-Basic_Generic_File_Failed.png)
</summary>

File: shared/chat/conversation/messages/wrapper.native.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  }
+}
+
+const _childrenWrapStyle = {
+  flex: 1,
+  ...globalStyles.flexBoxColumn,
+}
+
+const _stylesFirstNewMessage = {
+  borderTop: `solid 1px ${globalColors.orange}`,
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper.native.js]: this works?

**Comment by keybase-ci-visdiff** [comment]
The commits ebf7e3e6678c...082a30cd655a introduce visual changes on linux. <details><summary>:mag_right: 2 new, 7 changed</summary>

 * Added: **Text_Message-edited**  
   Text_Message-edited rendered
 * Added: **Stacked_Text_Message-Stacked_two_messages_one_edited**  
   Stacked_Text_Message-Stacked_two_messages_one_edited rendered
 * Changed: **Text_Message-failed_You** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Text_Message-failed_You-ebf7e3e6678c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Text_Message-failed_You-082a30cd655a.png)  
   Text_Message-failed_You comparison
 * Changed: **Text_Message-failed_Not_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Text_Message-failed_Not_Following-ebf7e3e6678c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Text_Message-failed_Not_Following-082a30cd655a.png)  
   Text_Message-failed_Not_Following comparison
 * Changed: **Text_Message-failed_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Text_Message-failed_Following-ebf7e3e6678c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Text_Message-failed_Following-082a30cd655a.png)  
   Text_Message-failed_Following comparison
 * Changed: **Text_Message-failed_Broken** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Text_Message-failed_Broken-ebf7e3e6678c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Text_Message-failed_Broken-082a30cd655a.png)  
   Text_Message-failed_Broken comparison
 * Changed: **Stacked_Text_Message-Stacked_one_sent_one_failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-ebf7e3e6678c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-082a30cd655a.png)  
   Stacked_Text_Message-Stacked_one_sent_one_failed comparison
 * Changed: **Attachment_Message-Basic_Preview_Image_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Attachment_Message-Basic_Preview_Image_Failed-ebf7e3e6678c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Attachment_Message-Basic_Preview_Image_Failed-082a30cd655a.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Attachment_Message-Basic_Preview_Image_Failed.png)
 * Changed: **Attachment_Message-Basic_Generic_File_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Attachment_Message-Basic_Generic_File_Failed-ebf7e3e6678c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Attachment_Message-Basic_Generic_File_Failed-082a30cd655a.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113320426-ebf7e3e6678c-082a30cd655a-linux/Attachment_Message-Basic_Generic_File_Failed.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits e9038c9d12bb...f3f82377d21c introduce visual changes on linux. <details><summary>:mag_right: 2 new, 7 changed</summary>

 * Added: **Text_Message-edited**  
   Text_Message-edited rendered
 * Added: **Stacked_Text_Message-Stacked_two_messages_one_edited**  
   Stacked_Text_Message-Stacked_two_messages_one_edited rendered
 * Changed: **Text_Message-failed_You** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Text_Message-failed_You-e9038c9d12bb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Text_Message-failed_You-f3f82377d21c.png)  
   Text_Message-failed_You comparison
 * Changed: **Text_Message-failed_Not_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Text_Message-failed_Not_Following-e9038c9d12bb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Text_Message-failed_Not_Following-f3f82377d21c.png)  
   Text_Message-failed_Not_Following comparison
 * Changed: **Text_Message-failed_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Text_Message-failed_Following-e9038c9d12bb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Text_Message-failed_Following-f3f82377d21c.png)  
   Text_Message-failed_Following comparison
 * Changed: **Text_Message-failed_Broken** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Text_Message-failed_Broken-e9038c9d12bb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Text_Message-failed_Broken-f3f82377d21c.png)  
   Text_Message-failed_Broken comparison
 * Changed: **Stacked_Text_Message-Stacked_one_sent_one_failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-e9038c9d12bb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-f3f82377d21c.png)  
   Stacked_Text_Message-Stacked_one_sent_one_failed comparison
 * Changed: **Attachment_Message-Basic_Preview_Image_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Attachment_Message-Basic_Preview_Image_Failed-e9038c9d12bb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Attachment_Message-Basic_Preview_Image_Failed-f3f82377d21c.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Attachment_Message-Basic_Preview_Image_Failed.png)
 * Changed: **Attachment_Message-Basic_Generic_File_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Attachment_Message-Basic_Generic_File_Failed-e9038c9d12bb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Attachment_Message-Basic_Generic_File_Failed-f3f82377d21c.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487113438634-e9038c9d12bb-f3f82377d21c-linux/Attachment_Message-Basic_Generic_File_Failed.png)
</summary>

**Comment by chrisnojima** [comment]
any idea why you still have the desktop visdiff?

**Comment by chromakode** [comment]
@chrisnojima aha, this is due to switching the retry component to flex. Will look at an alternative or add back the margins.

File: shared/chat/conversation/messages/wrapper.native.js (hunk 1 above)
Comment (chromakode) [inline shared/chat/conversation/messages/wrapper.native.js]: Fixed!

**Comment by keybase-ci-visdiff** [comment]
The commits a7614ea9c19e...c275312e2ce4 introduce visual changes on linux. <details><summary>:mag_right: 3 new, 7 changed</summary>

 * Added: **Text_Message-first_new_message**  
   Text_Message-first_new_message rendered
 * Added: **Text_Message-edited**  
   Text_Message-edited rendered
 * Added: **Stacked_Text_Message-Stacked_two_messages_one_edited**  
   Stacked_Text_Message-Stacked_two_messages_one_edited rendered
 * Changed: **Text_Message-failed_You** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Text_Message-failed_You-a7614ea9c19e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Text_Message-failed_You-c275312e2ce4.png)  
   Text_Message-failed_You comparison
 * Changed: **Text_Message-failed_Not_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Text_Message-failed_Not_Following-a7614ea9c19e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Text_Message-failed_Not_Following-c275312e2ce4.png)  
   Text_Message-failed_Not_Following comparison
 * Changed: **Text_Message-failed_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Text_Message-failed_Following-a7614ea9c19e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Text_Message-failed_Following-c275312e2ce4.png)  
   Text_Message-failed_Following comparison
 * Changed: **Text_Message-failed_Broken** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Text_Message-failed_Broken-a7614ea9c19e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Text_Message-failed_Broken-c275312e2ce4.png)  
   Text_Message-failed_Broken comparison
 * Changed: **Stacked_Text_Message-Stacked_one_sent_one_failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-a7614ea9c19e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-c275312e2ce4.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Stacked_Text_Message-Stacked_one_sent_one_failed.png)
 * Changed: **Attachment_Message-Basic_Preview_Image_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Attachment_Message-Basic_Preview_Image_Failed-a7614ea9c19e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Attachment_Message-Basic_Preview_Image_Failed-c275312e2ce4.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Attachment_Message-Basic_Preview_Image_Failed.png)
 * Changed: **Attachment_Message-Basic_Generic_File_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Attachment_Message-Basic_Generic_File_Failed-a7614ea9c19e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Attachment_Message-Basic_Generic_File_Failed-c275312e2ce4.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487263556443-a7614ea9c19e-c275312e2ce4-linux/Attachment_Message-Basic_Generic_File_Failed.png)
</summary>

**Comment by chromakode** [comment]
@chrisnojima This is ready for further review. I fixed the original visdiff, but there's still a one-pixel discrepancy. On desktop, this is the height of the retry line changing from 17px to 16px. I think the new, 16px height is more correct.

**Comment by keybase-ci-visdiff** [comment]
The commits 775b77b3ccad...4ca857405c77 introduce visual changes on linux. <details><summary>:mag_right: 3 new, 7 changed</summary>

 * Added: **Text_Message-first_new_message**  
   Text_Message-first_new_message rendered
 * Added: **Text_Message-edited**  
   Text_Message-edited rendered
 * Added: **Stacked_Text_Message-Stacked_two_messages_one_edited**  
   Stacked_Text_Message-Stacked_two_messages_one_edited rendered
 * Changed: **Text_Message-failed_You** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Text_Message-failed_You-775b77b3ccad.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Text_Message-failed_You-4ca857405c77.png)  
   Text_Message-failed_You comparison
 * Changed: **Text_Message-failed_Not_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Text_Message-failed_Not_Following-775b77b3ccad.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Text_Message-failed_Not_Following-4ca857405c77.png)  
   Text_Message-failed_Not_Following comparison
 * Changed: **Text_Message-failed_Following** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Text_Message-failed_Following-775b77b3ccad.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Text_Message-failed_Following-4ca857405c77.png)  
   Text_Message-failed_Following comparison
 * Changed: **Text_Message-failed_Broken** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Text_Message-failed_Broken-775b77b3ccad.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Text_Message-failed_Broken-4ca857405c77.png)  
   Text_Message-failed_Broken comparison
 * Changed: **Stacked_Text_Message-Stacked_one_sent_one_failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-775b77b3ccad.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Stacked_Text_Message-Stacked_one_sent_one_failed-4ca857405c77.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Stacked_Text_Message-Stacked_one_sent_one_failed.png)
 * Changed: **Attachment_Message-Basic_Preview_Image_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Attachment_Message-Basic_Preview_Image_Failed-775b77b3ccad.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Attachment_Message-Basic_Preview_Image_Failed-4ca857405c77.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Attachment_Message-Basic_Preview_Image_Failed.png)
 * Changed: **Attachment_Message-Basic_Generic_File_Failed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Attachment_Message-Basic_Generic_File_Failed-775b77b3ccad.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Attachment_Message-Basic_Generic_File_Failed-4ca857405c77.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487278217348-775b77b3ccad-4ca857405c77-linux/Attachment_Message-Basic_Generic_File_Failed.png)
</summary>

**Pull request merged**